emmarin/cl/ledger/src/covenant.rs

```diff
@@ -1,50 +1,48 @@
-use cl::cl::{Constraint, Nullifier, PtxRoot};
-use ledger_proof_statements::constraint::ConstraintPublic;
+use cl::crust::{Nullifier, TxRoot};
+use ledger_proof_statements::covenant::{SpendingCovenantPublic, SupplyCovenantPublic};
 
 use crate::error::Result;
 
 #[derive(Debug, Clone)]
-pub struct ConstraintProof {
+pub struct SupplyCovenantProof {
     pub risc0_id: [u32; 8],
     pub risc0_receipt: risc0_zkvm::Receipt,
 }
 
-pub fn risc0_constraint(risc0_id: [u32; 8]) -> Constraint {
-    unsafe { Constraint(core::mem::transmute::<[u32; 8], [u8; 32]>(risc0_id)) }
+#[derive(Debug, Clone)]
+pub struct SpendingCovenantProof {
```

Comment:
this is used for both burning and minting?

Reply:
this is for just for spending ,the other one yes, it's the same input for both burning and minting